script/make-release.js

```diff
@@ -0,0 +1,121 @@
+const fetch = require('node-fetch')
+
+const BASE_URL = 'https://api.github.com/repos/guardian/editions'
+
+const tagNameFromSha = sha => sha.slice(0, 8)
+
+const headers = {
+    'Content-Type': 'application/json',
+    Authorization: `token ${process.env.REPO_GITHUB_TOKEN}`,
+}
+
+const doFetch = async (url, params) => {
+    const resp = await fetch(url, params)
+    if (resp.status >= 200 && resp.status < 300) {
+        return resp.json()
+    } else {
+        console.log(
+            `Request to ${url} failed with status ${resp.status}: ${resp.statusText}. Exiting.`,
+        )
+        process.exit()
+    }
+}
+
+const get = async path => {
+    return doFetch(`${BASE_URL}/${path}`, { headers: headers })
+}
+
+const post = async (path, body) => {
+    return doFetch(`${BASE_URL}/${path}`, {
+        headers: headers,
+        method: 'POST',
+        body: JSON.stringify(body),
+    })
+}
+
+const patch = async (path, body) => {
+    return doFetch(`${BASE_URL}/${path}`, {
+        headers: headers,
+        method: 'PATCH',
+        body: JSON.stringify(body),
+    })
+}
+
+const findReleaseForCommit = (commit, releases) => {
+    return releases.find(r => r.tag_name == tagNameFromSha(commit))
+}
+
+const patchReleaseName = (name, os, appStoreId) =>
+    `${name}--${os}-${appStoreId}`
```

Comment:
Nitpick: we can use `:` for build number to indicate an association with the OS

The output would look like this:
`pm-github-release-script--ios:767--android:123456`

When we need to, this will also make it easy to parse out build number from release name